Subject: config.sub: accept substrate as an OS name

Add the substrate* OS family so --host=i386-unknown-substrate passes
configure's host-triple validation.

--- a/config/config.sub
+++ b/config/config.sub
@@ -2123,6 +2123,7 @@
 	| tops20* \
 	| tpf* \
 	| tvos* \
+	| substrate* \
 	| twizzler* \
 	| uclinux* \
 	| udi* \
